rotkehlchen/tests/fixtures/db.py

```diff
@@ -54,10 +55,11 @@ def fixture_session_data_dir(tmpdir_factory) -> Path:
 
 
 @pytest.fixture(name='user_data_dir')
-def fixture_user_data_dir(data_dir, username) -> Path:
+def fixture_user_data_dir(data_dir: Path, username: str, worker_id: str) -> Path:
     """Create and return the user data directory"""
-    user_data_dir = data_dir / username
-    user_data_dir.mkdir(exist_ok=True)
+    user_data_dir = data_dir / worker_id / username
+    shutil.rmtree(user_data_dir, ignore_errors=True)
```

Comment:
Why are you doing this? Isn't this why we have the `@pytest.mark.parametrize('use_clean_caching_directory', [True])` fixture?

Reply:
the reason to do `data_dir / worker_id / username` is that there are 3 parallel tests running and you can't have the same db, this is why to use the worker_id. As for the rmtree i got that if a test failed it couldn't run because the folder was populated but probably is due to how `use_clean_caching_directory` works. I'm going to check if this can be improved